### PR TITLE
Remove state from zip_with 

### PR DESCRIPTION
**Pull request by kaikalur**
We have this function still using state that's a problem for functions like REDUCE_AGG etc. So cleaning it up.

Test plan - tests exist

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by yingsu00 — COMMENTED** [review]
@kaikalur Could you enlighten me what problems the REDUCE_AGG has with the state?

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ZipWithFunction.java [hunk 1]
```diff
@@ -133,18 +119,13 @@ public static Block zipWith(
                 output = function.apply(left, right);
             }
             catch (Throwable throwable) {
-                // Restore pageBuilder into a consistent state.
-                arrayBlockBuilder.closeEntry();
```
[the remark this answers is not in the document]
Comment (kaikalur) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/ZipWithFunction.java:137]: I see no need as this is a local variable. Previously it was a block in the pagebuilder (the state that we just remvoed). So this also answers your other question. We simply don't want state in scalar functions as it is trouble. Search in the issues - I filed a couple of issues with this.

**Comment by kaikalur** [comment]
> @kaikalur Could you enlighten me what problems the REDUCE_AGG has with the state?

Replied inline. Also see:

https://github.com/prestodb/presto/issues/16728

We started with that and now finally removed it from all scalars.

**Comment by kaikalur** [comment]
Hello, can one of the committers merge this cleanup fix asap? Thanks!

**Comment by kaikalur** [comment]
> I understand why we want to get rid of the state, but what value was it providing before? Anything to watch out for with this change?

It was supposed to be for memory performance by reducing the number of block builders created etc. but we noticed in practice it is actually not helping and were fixing a lot of array/map scalar functions but this one fell through the cracks. So I'm fixing it now. See the numbers below in a previous PR:

https://github.com/prestodb/presto/pull/16734/commits/fb810919f8b22ad75b260620aac92275261f4695

IMO, irrespective of that, this is a sound change because we simply don't want state in scalars.

**Comment by rschlussel** [comment]
This fixes a possible correctness issue, right?  Can you add a release note.

**Pull request merged**